docs/_ext/_custom_icons.py

```diff
@@ -0,0 +1,58 @@
+"""'In-tree' sphinx extension to add icons/favicons to documentation"""
```

Comment:
I'd like to see this as a separate package so it doesn't have to be injected. Let me know if that's something you'd like to do. If not, I can do it.

Reply:
I think it's fine for now. This is a common approach. Later, it could be implemented upstream instead.

Reply:
I already am working together with the maintainer of [`sphinx-favicon`](https://github.com/tcmetzger/sphinx-favicon) to port the same functionality implemented here to that package 😄 

Maybe in a few days?

If that, works, do you guys prefer to have `sphinx-favicon` as a doc dependency instead?

Reply:
Yes, that would be great. Any reason not to push this all the way through to the Sphinx upstream (not an extension)?

Reply:
Sounds good. Since it sounds like the behavior will be available externally at some point, I'm happy to live with the vendored behavior for a few weeks.